src/blk/kernel/KernelDevice.cc

```diff
@@ -540,10 +539,25 @@ void KernelDevice::_aio_stop()
 {
   if (aio) {
     dout(10) << __func__ << dendl;
+    int pipe_fd[2];
+    bool pipe_created = (pipe(pipe_fd)==0);
+    IOContext wakeup_ctx(cct,nullptr,false);
+    wakeup_ctx.num_running = 1;
```

Comment:
Is setting num_running important? I thought it is a thing that is incremented in `submit_batch` and decreased when we retrieve op completion.

Reply:
Ok, I see it now.
We normally ++ it in `KernelDevice::aio_write()`, which we cannot use here.